agent/agent-tooling/src/main/java/io/opentelemetry/sdk/metrics/internal/view/MetricViewAttributesProcessor.java

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.opentelemetry.sdk.metrics.internal.view;
+
+import static com.azure.monitor.opentelemetry.exporter.implementation.AiSemanticAttributes.IS_PRE_AGGREGATED;
+
+import com.azure.monitor.opentelemetry.exporter.implementation.AiSemanticAttributes;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+@SuppressWarnings("rawtypes")
+class MetricViewAttributesProcessor extends AttributesProcessor {
+
+  private final Set<AttributeKey<?>> includeAttributeKeys;
+  private final boolean includeSynthetic;
+
+  MetricViewAttributesProcessor(
+      Set<AttributeKey<?>> includeAttributeKeys, boolean includeSynthetic) {
+    this.includeAttributeKeys = includeAttributeKeys;
+    this.includeSynthetic = includeSynthetic;
+  }
+
+  @Override
+  public Attributes process(Attributes incoming, Context context) {
+
+    Span span = Span.fromContext(context);
+
+    // this is needed for detecting telemetry signals that will trigger pre-aggregated metrics via
+    // auto instrumentations
+    span.setAttribute(AiSemanticAttributes.IS_PRE_AGGREGATED, true);
+
+    AttributesBuilder filtered = Attributes.builder();
+    applyCommon(filtered, span);
+    applyView(filtered, incoming, includeAttributeKeys);
+    if (includeSynthetic) {
+      filtered.put(AiSemanticAttributes.IS_SYNTHETIC, UserAgents.isBot(incoming));
+    }
+    return filtered.build();
+  }
+
+  @Override
+  public boolean usesContext() {
+    return true;
+  }
+
+  private static void applyCommon(AttributesBuilder builder, Span span) {
+
+    // this is needed for detecting telemetry signals that will trigger pre-aggregated metrics via
+    // auto instrumentations
+    span.setAttribute(IS_PRE_AGGREGATED, true);
+
+    if (!(span instanceof ReadableSpan)) {
+      return;
+    }
+
+    ReadableSpan readableSpan = (ReadableSpan) span;
+
+    String connectionString = readableSpan.getAttribute(AiSemanticAttributes.CONNECTION_STRING);
+    if (connectionString != null) {
+      builder.put(AiSemanticAttributes.CONNECTION_STRING, connectionString);
```

Comment:
why do we need this for pre-agg?

Reply:
so pre-aggs get reported to the correct connection string when `connectionStringOverrides` are used

Reply:
can we add a comment there mentioning connecitonStringOverrides?